src/base/reports/IReporter.hpp

```diff
@@ -34,8 +34,12 @@ struct IODelayReport {
 
 class IReporter {
 public:
+    IReporter() = default;
     virtual ~IReporter() = default;
 
+    IReporter(const IReporter&) = delete;
+    IReporter& operator=(const IReporter&) = delete;
```

Comment:
Do you think we really need to do this for interfaces?

In theory, an interface doesn't know whether its implementations are copyable or not?

Reply:
I can also move it back to the derived child classes.

I just moved it here into the base class to 1.) avoid code duplication and 2.) classes in polymorphic hierarchies very rarely implement public copy/move operations. Usually they are either declared private or protected in the base class in order to avoid slicing, see also [C++-Core Guideline C.130](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-copy)

Reply:
I didn't think about it this way. I agree, it makes sense.